addons/movments_accounts_balances/controllers/controllers.py

```diff
@@ -97,108 +98,663 @@ def general_ledger_report(self, account_id, start_date, end_date):
+        if not invoice:
+            return {'error': 'Bill not found'}
+
+        # Prepare data for invoice lines
+        invoice_line_data = [{
+            'line_id': line.id,
+            'account_id': line.account_id.id,
+            'account_name': line.account_id.name,
+            'quantity': line.quantity,
+            'price_unit': line.price_unit,
+        } for line in invoice.invoice_line_ids]
+
+        # Assemble bill data
+        invoice_data = {
+            'id': invoice.id,
+            'bill_number': invoice.ref,
+            'bill_date': invoice.invoice_date,
+            'supplier_id': invoice.partner_id.name,
+            'amount': invoice.amount_total,
+            'state': invoice.payment_state,
+            'invoice_lines': invoice_line_data,
+        }
+
+        return {'invoice_info': invoice_data}
+
+    @api.model
+    def delete_bill(self, bill_id):
+        Bill = self.env['account.move']
+        bill = Bill.search([('id', '=', bill_id), ('move_type', '=', 'in_invoice')])
+
+        if not bill:
+            return "Bill not found."
+
+        # Check if the bill is posted (validated)
+        if bill.state == 'posted':
+            # Add logic to cancel related journal entries (if any)
+            # Example: bill.button_draft() or bill.button_cancel()
+            try:
+                bill.button_draft()  # Reset to draft
+            except Exception as e:
+                return "Failed to reset bill to draft: {}".format(e)
+
+        # Additional logic to unreconcile payments if the bill is reconciled
+
+        try:
+            bill.unlink()  # Delete the bill
+            return "Bill deleted successfully."
+        except Exception as e:
+            return "Failed to delete bill: {}".format(e)
+
+    ##create/get/delete_bills payment
+    @api.model
+    def create_bill_payment(self, invoice_vals, partner_id, invoice_date, invoice_date_due, ref, narration):
+        """
+        Create a payment based on the provided data.
+
+        :param invoice_vals: List of dictionaries containing line data for the payment.
+        :param partner_id: Partner ID for the payment.
+        :param invoice_date: Invoice date for the payment.
+        :param invoice_date_due: Due date for the payment.
+        :param ref: Reference for the payment.
+        :param narration: Narration for the payment.
+        :return: The created payment record.
+        """
+        # Create a new payment record
+        payment = self.env['account.payment'].create({
+            'move_type': 'in_invoice',
+            'partner_id': partner_id,
+            'invoice_date': invoice_date,
+            'invoice_date_due': invoice_date_due,
+            'ref': ref,
+            'narration': narration,
+            'invoice_line_ids': [(0, 0, {
+                'name': line.get('description', ''),
+                'price_unit': line.get('amount', 0.0),
+                'account_id': line.get('account_id'),
```

Comment:
There is no account_id for create_bill_payment. Did the method in my PR not work? https://github.com/sugampandey/odoo/pull/12/files